lang/query/src/lib.rs

```diff
@@ -6,62 +9,133 @@ use rust_lapper::Lapper;
 pub use result::Error;
 
 mod asserts;
+mod cache;
+mod edit;
+mod fs;
 mod info;
+mod lift;
+mod load;
+mod modules;
 mod result;
-mod view;
-mod view_mut;
+mod rt;
+mod spans;
+mod xfunc;
+
+use cache::*;
 
+pub use edit::*;
+pub use fs::*;
 pub use info::*;
-pub use view::*;
-pub use view_mut::*;
+pub use load::*;
+pub use xfunc::*;
 
-#[derive(Default)]
+/// A database tracking a set of source files
 pub struct Database {
```

Comment:
Alternative design: `Database<T: FileSource>` and `source: Box<T>` instead of  `source: Box<dyn FileSource>`. This would show in the API which functions are generic or only make sense for one representation.